### ISSUE TITLE
Using arrow keys on tab completion (modify item) crashes the application

### ISSUE THREAD

**Issue by Petemir** [kdheepak/taskwarrior-tui#177]
If I try to use arrows for the selection of the appropriate item in the "Tab Completion" pop-up when modifiying an item, the application crashes. I know that it is possible to use tab/shift-tab for navigation.

* Open taskwarrior-tui
* Select any task and click `m` for modification
* Touch tab
* Touch any vertical arrow (up/down)

**Environment (please complete the following information):**

- Operating System: Linux (Ubuntu 20.04)
- Installation: GitHub releases (AppImage)
- taskwarrior-tui version: v0.13.7
- task version: 2.5.1

**Additional context and information**

The error is the following:
```The application panicked (crashed).
  assertion failed: pos <= buf.len()
in /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rustyline-8.0.0/src/line_buffer.rs, line 151```

If needed, let me know and I'll run it with the full backtrace.

**Comment by kdheepak** [comment]
Thanks for reporting! I fixed this in one spot but not others. 

Also, using `Up` and `Down` currently navigate history of the filter OR add / modify / log prompts. I've made a PR that changes the behavior of `Up` and `Down` when the tab completion pane is open. The history feature I think is something that needs some polish. And ideally, the tab completion pane should show history prompts as well (which is currently does not do).

All that is to say tab completion and history at very much in alpha at the moment, and I have plans to make improvements there. Feel free to report any more bugs you come across or feature requests you'd like!
